x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/analysis/VerifierTests.java

```diff
@@ -58,8 +81,280 @@ public void testMisspelledColumnWithMultipleOptions() {
+        assertEquals("1:11: Cannot use field [integer_range_field] with unsupported type [integer_range]",
+                error(idxr, "foo where integer_range_field == ''"));
+        assertEquals("1:11: Cannot use field [float_range_field] with unsupported type [float_range]",
+                error(idxr, "foo where float_range_field == ''"));
+        assertEquals("1:11: Cannot use field [long_range_field] with unsupported type [long_range]",
+                error(idxr, "foo where long_range_field == ''"));
+        assertEquals("1:11: Cannot use field [double_range_field] with unsupported type [double_range]",
+                error(idxr, "foo where double_range_field == ''"));
+        assertEquals("1:11: Cannot use field [date_range_field] with unsupported type [date_range]",
+                error(idxr, "foo where date_range_field == ''"));
+        assertEquals("1:11: Cannot use field [ip_range_field] with unsupported type [ip_range]",
+                error(idxr, "foo where ip_range_field == ''"));
+    }
+
+    public void testObject() {
+        final IndexResolution idxr = loadIndexResolution("mapping-object.json");
+        accept(idxr, "foo where endgame.pid == 0");
+
+        assertEquals("1:11: Unknown column [endgame.pi], did you mean [endgame.pid]?",
+                error(idxr, "foo where endgame.pi == 0"));
+
+    }
+
+    public void testNested() {
+        final IndexResolution idxr = loadIndexResolution("mapping-nested.json");
+        accept(idxr, "foo where processes.pid == 0");
+
+        assertEquals("1:11: Unknown column [processe.pid], did you mean any of [processes.pid, processes.path, processes.path.keyword]?",
+                error(idxr, "foo where processe.pid == 0"));
+    }
+
+    public void testGeo() {
+        final IndexResolution idxr = loadIndexResolution("mapping-geo.json");
+        assertEquals("1:11: Cannot use field [location] with unsupported type [geo_point]",
+                error(idxr, "foo where location == 0"));
+        assertEquals("1:11: Cannot use field [site] with unsupported type [geo_shape]",
+                error(idxr, "foo where site == 0"));
+    }
+
+    public void testIP() {
+        final IndexResolution idxr = loadIndexResolution("mapping-ip.json");
+        accept(idxr, "foo where ip_addr == 0");
+    }
+
+    public void testJoin() {
+        final IndexResolution idxr = loadIndexResolution("mapping-join.json");
+        accept(idxr, "foo where serial_event_id == 0");
+    }
+
+    public void testMultiField() {
+        final IndexResolution idxr = loadIndexResolution("mapping-multi-field.json");
+        accept(idxr, "foo where multi_field.raw == 'bar'");
+
+        assertEquals("1:11: [multi_field.english == 'bar'] cannot operate on first argument field of data type [text]: " +
+                        "No keyword/multi-field defined exact matches for [english]; define one or use MATCH/QUERY instead",
```

Comment:
I think this error message is wrong, unless EQL plans to support MATCH/QUERY predicates.

Reply:
@costin

Reply:
@aleksmaus please raise an issue and mark it as discuss to get consensus about how we deal with exact matches on analyzed fields - whether we prevent any exact comparison against them or introduce some other type of comparison (inexact matches - say `~=`).